packages/@lwc/ssr-compiler/src/compile-template/transformers/text.ts

```diff
@@ -32,7 +32,7 @@ const bYieldEscapedString = esTemplateWithYield`
             yield String(${0});
             break;
         default:
-            yield ${0} ? htmlEscape(${0}.toString()) : '\\u200D';
+            yield ${0} ? htmlEscape(${0}.toString()) : ${2} ? '\\u200D' : '';
```

Comment:
This was not strictly related to the `@wire` stuff, but was necessary in order to get the `context-no-provider` test to pass, where this happens in the template: `<div>I have no context for this{foo}</div>`.

Prior to my change here, a `\u200D` character would be inserted following the word "this" in the rendered markup. That's not what we want - the zero-width character should only be rendered if there are no adjacent text nodes. This check happens in `case 'string'` a few lines above.

Reply:
Yeah we have [an open WI](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000024NgLoYAK/view) to fix adjacent-text-concatenation in SSR.